### PR TITLE
Implement view Functionality

### PR DESCRIPTION
**Pull request by Zafron047**
## Dear Reviewer,
This pull request lays the foundation for the project's user interface and functionality. Styling enhancements will be addressed in subsequent phases, Please review this pull request and provide feedback or approve to merge and proceed to next part of this project.

**Summary:**
This pull request addresses the implementation of the view for the project. The primary focus has been on ensuring that all necessary functionality is in place.
**As styling being considered as a nice-to-have but secondary concern at this stage. So I have used basic bootstrap styling.**

**Details:**

**View Updates:** The user interface has undergone substantial updates. The following changes have been made:
   - In the `users/index.html.erb` template, all users are now displayed.
   - The `users/show.html.erb` template now displays specific user details.
   - In the `posts/index.html.erb` template, all posts are listed.
   - The `posts/show.html.erb` template now showcases specific post information.
   - Counters for comments, posts, and likes have been introduced for a more comprehensive view.

**Review Checklist:**
- [x] Wireframes have been implemented.
- [x] Core functionality is working as intended.
- [x] Data processing methods are used effectively.
- [x] Data seeding has been performed for posts and comments.
- [x] Styling enhancements will be addressed in subsequent milestones.

Thank you in advance for your review and feedback!

Regards,

@Zafron047

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,18 @@
...
+          <div class="card-body" style="border: 1px solid black; margin-top: 5px;">
+            <h5 class="card-title"><%= link_to user.name, user_path(user), style: "text-decoration: none;" %></h5>
+            <p class="card-text"><small class="text-muted">Number of posts: <%= user.posts_counter %></small></p>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+
+<% end %>
```
Comment (DammyShittu) [inline app/views/users/index.html.erb:18]: - **[OPTIONAL]** It's often the case that we use partials to render collections of this kind in Rails. You can also do the same for better clean up of the view.

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -1 +1,27 @@
...
+        </div>
+        <div>
+          <span class="card-text">Comments: <%= @post.comments_counter%></span>
+          <span class="card-text">Likes: <%= @post.likes_counter%></span>
+        </div>
+      </div>
+      <p class="card-text"><%= @post.text %></p>
+    </div>
+  </div>
+  <% if @recent_comments.any? %>
```
Comment (DammyShittu) [inline app/views/posts/show.html.erb]: - [ ] The post show page should render all the comments of a post, not the recent comments. That should be done in the post index page.

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,46 @@
...
+        </a>
+        <p class="card-text">
+          <%= post.text %>
+        </p>
+        <div class= "card-text" style="float: right">
+          <span class="card-text">Comments: <%= post.comments_counter%></span>
+          <span class="card-text">Likes: <%= post.likes_counter%></span>
+        </div>
+      </div>
+      <% if post.comments.any? %>
```
Comment (DammyShittu) [inline app/views/posts/index.html.erb:31]: - [ ] The comments that are rendered on the post index page should be the recent comments, not all comments. Kindly fix this.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Zafron047,

## Highlights
✔️ Good layout designs

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 3 above)
Comment (Zafron047) [reply app/views/posts/index.html.erb:31]: I have fixed this issue and recent comments are now visible in the post index page:

5most_recent_comments

File: app/views/posts/show.html.erb (hunk 2 above)
Comment (Zafron047) [reply app/views/posts/show.html.erb]: I have fixed this issue. All comments related with the post is now visible in the post show page:

all_comments_showjpg

**Comment by Zafron047** [comment]
@DammyShittu Thank you for finding out the issues I have fixed the views as you have asked.

all_comments_showjpg

5most_recent_comments

File: app/views/posts/show.html.erb (hunk 2 above)
Comment (codepantha) [inline app/views/posts/show.html.erb]: - [ ] On the posts show page, you are still showing the recent comments just like the previous reviewer highlighted. Please ensure to show all the comments here 👍 
From the attached screenshot below, you can see that the comment count is 6, but you are only displaying 4

image

File: app/views/posts/index.html.erb (hunk 3 above)
Comment (codepantha) [inline app/views/posts/index.html.erb:31]: https://github.com/Zafron047/Blog_app_by_Rails/blob/4d771a9d4026028ac9027b89d2b3fc2798a5441e/app/models/post.rb#L22-L24
- [ ] Well done for showing some recent comments 👏, however, in the project requirements, you are expected to render a post's **5 most recent comments** on the posts index page but you are only rendering 4. I noticed that you tweaked your method to allow users to set the limit, however, please set the default `limit = 5` 👍 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Zafron047 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ Stuck with wireframe layouts 📌 
✔️ Added views associated with the models and controllers 📌 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 3 above)
Comment (Zafron047) [reply app/views/posts/index.html.erb:31]: I have also removed the extra loop that was rendering all the comments after loading the 5 recent comments.

5most_recent_comments

**Comment by Zafron047** [comment]
@codepantha Thanks a lot for the slack huddle and debug rendering the all post relevant comments in the show page.

I have also fixed the 5recent comments + all comments from rendering in all comments. I appreciate your kind effort.

Thanks a lot. ^_^

File: app/views/posts/show.html.erb (hunk 2 above)
Comment (Zafron047) [reply app/views/posts/show.html.erb]: Thanks a lot for the walk through and debugging. Now it is showing only the relevant comments.

all_comments_showjpg

**Review by Gambit142 — COMMENTED** [review]
Hi @Zafron047  👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- Setup linters correctly ✔️ 
- Created Index and Show pages for Users ✔️ 
- Created Index and Show pages for Posts ✔️

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Zafron047** [comment]
I have already merged a while ago after you approved. However, thank you for adding the approval message here also.

Have a great day. 😃

**Pull request merged**